### ISSUE TITLE
Custom 404 page stops working when using pathPrefix

### ISSUE THREAD

**Issue by hannesd** [gatsbyjs/gatsby#15832]
## Description

When navigating to a non-existent path the 404 page flashes and disappears, and I see an empty page.

This only happens when I use a pathPrefix

### Steps to reproduce

`$ gatsby new`

Choose a project name of your liking.
Choose the "gatsby-starter-hello-world".

Create a custom 404 page at `src/pages/404.js`:
```
import React from "react"

export default () => <div>404</div>
```

`$ rm .cache public -rf && npm run build && npm run serve`

Navigate to http://localhost:9000

You should see a page containing "Hello World"

Navigate to http://localhost:9000/asdf

You should see a page containing "404"

Configure a pathPrefix:

`package.json`:
```
{
  ...
  "scripts": {
    ...
    "build": "gatsby build --prefix-paths",
    ...
    "serve": "gatsby serve --prefix-paths",
    ...
  },
  ...
}
```

`gatsby-config.js`:
```
module.exports = {
  pathPrefix: '/foo',
}
```

`$ rm .cache public -rf && npm run build && npm run serve`

Navigate to http://localhost:9000/foo

You should see a page containing "Hello World"

Navigate to http://localhost:9000/foo/asdf

You should see a page containing "404" but you will get an empty page.

### Expected result

I would expect to see the 404 page with the "404" content for every non-existent path below the pathPrefix.

### Actual result

I see an empty page. Sometimes I can see the 404 page flash before it disappears.

### Environment

```
  System:
    OS: Linux 4.18 Ubuntu 18.10 (Cosmic Cuttlefish)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
    Shell: 4.4.19 - /bin/bash
  Binaries:
    Node: 11.15.0 - /usr/bin/node
    npm: 6.9.0 - ~/.npmprefix/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.100
    Firefox: 68.0
  npmPackages:
    gatsby: ^2.13.25 => 2.13.25
  npmGlobalPackages:
    gatsby-cli: 2.7.15
```

### Related Issues

I have found similar sounding issues but they were refering to the offline plugin and have been closed a long time ago.

**Comment by pieh** [comment]
I definitely can reproduce it. Thanks for report - I'll make sure to fix this and add tests to catch regressions with 404 and path prefix in future while doing so

**Comment by pieh** [comment]
I opened pull request with fix for this https://github.com/gatsbyjs/gatsby/pull/15858 ( if you want to track progress on this bugfix for this)

**Comment by hannesd** [comment]
Thanks!

**Comment by sidharthachatterjee** [comment]
Fix published in `gatsby@2.13.28`
